cmd/goal/application.go

```diff
@@ -1365,8 +1408,12 @@ var methodAppCmd = &cobra.Command{
 				reportErrorf(err.Error())
 			}
 
+			if methodCreation && resp.ApplicationIndex != nil && *resp.ApplicationIndex != 0 {
+				reportInfof("Created app with app index %d", *resp.ApplicationIndex)
+			}
```

Comment:
minor question: do we need to concern the existence of `*resp.ApplicationIndex` on the condition of:
- `client.PendingTransactionInformationV2` successfully responded with no error
- creating an application from app-call transaction

If not, it seems okay to lift the check of null-ptr?

Reply:
I'd rather have the specific checks in an if statement than assume the pointer will always be valid, since you never know what could change in the future.

For what it's worth, the existing create app command has a similar check too: https://github.com/algorand/go-algorand/blob/2e09ebfca1ca46231e43cd62c48a1025ce7c0a11/cmd/goal/application.go#L490-L492